### DISCUSSION TITLE
Multiple App Instances on One Page - Zustand Store Shared between instances?

### DISCUSSION THREAD

**Discussion by lawskiddle** [pmndrs/zustand#2637] [Q&A]
Hi,

We're currently using `zustand` on an app that can be embedded on any page multiple times. 

While we want the `zustand` store to be shared throughout the components within the app, which is standard `zustand` behaviour, when **multiple instances of the app** are placed on a page without iframes (the app is a checkout-style component) the `zustand` stores of the different apps interfere with one another. I presume this is due to the same global state being used and manipulated by the different instances of the app?

How can we scope the `zustand` stores, so this kind of behaviour doesn't occur?

**Comment by dbritto-dev** [comment]
@lawskiddle you can do that using different contexts with Context API
